### PR TITLE
fix: Bookings Page goes blank on reschedule

### LINKED ISSUE #16422: /booking: after taking an action the page goes blank
after i click "Request for reschedule" the page goes blank:

CleanShot 2024-08-30 at 12 04 29@2x

not sure if its the same with cancel

### PR DESCRIPTION
**Pull request by chaithanya009**
## What does this PR do?

- Fixes #16422 (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

In bookings page when we cancel a booking using "request on reschedule" the page goes blank after few bookings are cancelled.
image

This issue is addressed and now only the cancelled booking is removed from UI

### CODE REVIEW

File: apps/web/components/dialog/RescheduleDialog.tsx [hunk 1]
```diff
@@ -23,15 +24,17 @@ interface IRescheduleDialog {
...
+  const queryClient = useQueryClient();
 
   const { mutate: rescheduleApi, isPending } = trpc.viewer.bookings.requestReschedule.useMutation({
     async onSuccess() {
       showToast(t("reschedule_request_sent"), "success");
       setIsOpenDialog(false);
-      await utils.viewer.bookings.invalidate();
+      const queryKeys = getQueryKey(trpc.viewer.bookings.get, undefined, "infinite");
+      queryClient.removeQueries({ queryKey: queryKeys });
+      queryClient.invalidateQueries(queryKeys);
```
Comment (anikdhabal) [inline apps/web/components/dialog/RescheduleDialog.tsx:37]: This is not correct

**Review by anikdhabal — CHANGES_REQUESTED** [review]

**Comment by chaithanya009** [comment]
The underlying issue is the cursor for the infinite query used in bookings is not reset on invalidation.
refer: https://tanstack.com/query/latest/docs/framework/react/guides/infinite-queries?from=reactQueryV3#what-happens-when-an-infinite-query-needs-to-be-refetched

**Review by Amit91848 — COMMENTED** [review]
Can you add a before after video? Can't seem to replicate it 🤔 @chaithanya009

**Comment by anikdhabal** [comment]
> The underlying issue is the cursor for the infinite query used in bookings is not reset on invalidation. refer: https://tanstack.com/query/latest/docs/framework/react/guides/infinite-queries?from=reactQueryV3#what-happens-when-an-infinite-query-needs-to-be-refetched

First, I can't reproduce the issue on my end. Could you please share a Loom video of the bug and its solution?

**Comment by chaithanya009** [comment]
Closing the PR. From what I tested there is an issue invalidating infiniteQuery.

**Pull request closed without merging**